Centaurus.DAL/Mongo/MongoStorage.cs

```diff
@@ -340,16 +327,51 @@ public async Task Update(DiffObject update)
                     }
                     catch (Exception exc)
                     {
-                        tries++;
                         if (tries <= maxTries)
                         {
-                            logger.Warn(exc, $"Error during update. {tries} try.");
+                            logger.Debug(exc, $"Error during update. {tries} try.");
+                            tries++;
                             continue;
                         }
                         new Exception($"Unable to commit transaction after {tries} tries", exc);
                     }
                 }
             }
+            return tries;
+        }
+
+        private void SaveQuanta(ref List<Task> updateTasks, IEnumerable<QuantumModel> quanta, IClientSessionHandle session)
+        {
+            var batchSize = 10000;
+
+            var savedQuantaCount = 0;
+            var quantaCount = quanta.Count();
+            while (savedQuantaCount < quantaCount)
+            {
+                updateTasks.Add(quantaCollection.InsertManyAsync(
+                    session,
+                    quanta.Skip(savedQuantaCount).Take(batchSize),
+                    new InsertManyOptions { BypassDocumentValidation = true, IsOrdered = false })
+                );
+                savedQuantaCount += batchSize;
+            }
+        }
+
+        private void SaveEffects(ref List<Task> updateTasks, IEnumerable<EffectsModel> effects, IClientSessionHandle session)
+        {
+            var batchSize = 1000;
```

Comment:
```suggestion
            var batchSize = 4000;
```
I think it will be better that way